### PR TITLE
Add total_paid field on order

### PR DESCRIPTION
**Pull request by tomaszszymanski129**
I want to merge this change because it adds `total_paid` field on order to limit payments queries when retrieveing some order - payment related stuff.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3113 order.total_paid optimization.](https://app.clickup.com/t/2549495/SALEOR-3113)

File: saleor/graphql/order/mutations/orders.py [hunk 1]
```diff
@@ -150,6 +150,9 @@ def try_payment_action(order, user, payment, func, *args, **kwargs):
         raise ValidationError(
             {"payment": ValidationError(message, code=OrderErrorCode.PAYMENT_ERROR)}
         )
+    else:
```
Comment (korycins) [inline saleor/graphql/order/mutations/orders.py:153]: Not sure if `else` is needed here, errors will raise ValidationError, so you can update total without `else` section.

File: saleor/payment/utils.py [hunk 2]
```diff
@@ -308,6 +308,8 @@ def gateway_postprocess(transaction, payment):
         payment.save(update_fields=changed_fields)
     transaction.already_processed = True
     transaction.save(update_fields=["already_processed"])
+    if order := payment.order:
+        order.update_total_paid()
```
Comment (korycins) [inline saleor/payment/utils.py:312]: It adds an additional query for saving the captured_amount, event if we don't modify it. Why don't you check if `captured_amount in changed_fields`  and the call update_total_paid?

File: saleor/graphql/order/mutations/orders.py (hunk 1 above)
Comment (korycins) [reply saleor/graphql/order/mutations/orders.py:153]: Also, can you double check if you really need this here? It is a wrapper for functions from gateway.py. All actions like capture, refund, etc - are decorated with `gateway_postprocess` which already call `order.update_total_paid`

**Review by korycins — APPROVED** [review]
Just to confirm, have you tested payment methods with different payment gateways than dummy?

**Comment by tomaszszymanski129** [comment]
Will smoke test it with Adyen

**Comment by tomaszszymanski129** [comment]
Everything works as expected, tested with Adyen and testing card

**Review by fowczarek — APPROVED** [review]

**Review by eCzerniak — APPROVED** [review]

**Pull request merged**